pkg/core/interfaces.go

```diff
@@ -131,4 +131,6 @@ type ExecutionManager interface {
 type Requests interface {
 	// MakeAPIRequest makes an HTTP request
 	MakeAPIRequest(ctx context.Context, httpMethod, endpoint string, body []byte) ([]byte, error)
+	// MakeAPIRequest makes an HTTP request with auth
+	MakeAPIRequestWithAuth(ctx context.Context, httpMethod, endpoint string, body []byte) ([]byte, error)
```

Comment:
instead of creating this function (and implementing it), you can extend the existing `MakeAPIRequest` function to take query-params and headers as additional inputs.

Reply:
yes